crates/shared/src/oneinch_api.rs

```diff
@@ -496,7 +503,12 @@ impl OneInchClientImpl {
     pub const SUPPORTED_CHAINS: &'static [u64] = &[1, 100];
 
     /// Create a new 1Inch HTTP API client with the specified base URL.
-    pub fn new(base_url: impl IntoUrl, client: Client, chain_id: u64) -> Result<Self> {
+    pub fn new(
+        base_url: impl IntoUrl,
+        client: Client,
+        chain_id: u64,
+        block_retriever: Arc<dyn BlockRetrieving>,
```

Comment:
I think we should pass around `CurrentBlockStream` everywhere instead of `dyn BlockRetrieving`. This would result in way fewer calls to the node because `CurrentBlockStream` can simply return whatever value it thinks is current (also reduced latency across the board).
Additionally this would give a single "heartbeat" for the entire backend (single source of truth regarding current block).

Reply:
I wasn't sure how to get the latest block from the CurrentBlockStream. It allows to stream new blocks but has no convenient way of getting the latest one (I guess we could locally store the latest one and then on each try `stream.next().now_or_never()` until it fails). Maybe this itself could be a BlockRetrieving implementation?

I'm also a bit worried about the extra roundtrip to the node we are now adding on every price estimate (I was thinking of caching block info in the BlockRetriever if this became necessary)

Reply:
You can do `let current_block = self.current_block.borrow().number;` with a `CurrentBlockStream`.